**Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):  
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],    
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].  
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.**

In [3]:
import numpy as np
import pandas as pd
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

# Находим среднее значение для каждой выборки
zp_mean = sum(zp) / len(zp)
ks_mean = sum(ks) / len(ks)

# Находим ковариацию
cov = sum((zp[i] - zp_mean) * (ks[i] - ks_mean) for i in range(len(zp))) / len(zp)

print("Ковариация:", cov)

Ковариация: 9157.84


In [4]:
# Находим ковариацию с помощью функции cov из библиотеки numpy
cov = np.cov(zp, ks, ddof=0)[0][1]

print("Ковариация:", cov)

Ковариация: 9157.84


In [5]:
# Находим коэффициент корреляции Пирсона с помощью функций из библиотек numpy и pandas.
r = np.corrcoef(zp, ks)
print(r)

[[1.         0.88749009]
 [0.88749009 1.        ]]


In [6]:
df = pd.DataFrame({'zp': zp, 'ks': ks})
r = df.corr()
print(r)

         zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000


Ответ: 
Ковариация двух величин равна 9157,84
Коэффициент корреляции Пирсона двух величин равен 0,88749009

**Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:  
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.  
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95** 


In [7]:
data = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
mean = np.mean(data)
std = np.std(data, ddof=1)

In [9]:
# Найдем t-значение для интервала с надежностью 0.95 и 9 степенями свободы (10 минус 1):

t = 2.262
n = len(data)
lower = mean - t * std / np.sqrt(n)
upper = mean + t * std / np.sqrt(n)
print(lower, upper)

110.55660776308164 125.64339223691834


Ответ: доверительный интервал располагается в диапазоне (110.56, 125,64)

**Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.**

In [13]:
import scipy
alpha = 0.05
n = 27
mean = 174.2
std = np.sqrt(25)

t = np.abs(np.round(np.float64(
    np.abs(
        np.round(
            scipy.stats.t.ppf(alpha/2, df=n-1), 3
        )
    )
), 3))

print("t-критерий Стьюдента:", t)

t-критерий Стьюдента: 2.056


In [14]:
interval = np.round(np.float64([
    mean - t * std / np.sqrt(n),
    mean + t * std / np.sqrt(n)
]), 2)

print("Доверительный интервал:", interval)

Доверительный интервал: [172.22 176.18]


Ответ: 
Доверительный интервал располагается в диапазоне  (172.22 176.18])